sdks/java/core/src/main/java/org/apache/beam/sdk/runners/PTransformMatcher.java

```diff
@@ -33,6 +33,10 @@
 public interface PTransformMatcher {
   boolean matches(AppliedPTransform<?, ?, ?> application);
 
+  default boolean matchesDuringValidation(AppliedPTransform<?, ?, ?> application) {
```

Comment:
Add a comment about why this method was necessary.

Reply:
That's a good suggestion. Thanks!